primitives/src/targeting/eval_test.rs

```diff
@@ -0,0 +1,1346 @@
+            )
+        ];
+
+        for (string, separator, expected) in cases.iter() {
+            let rule = Rule::Function(Function::new_split(
+                Rule::Value(string.clone()),
+                Rule::Value(separator.clone()),
+            ));
+
+            assert_eq!(Ok(Some(expected.clone())), rule.eval(&input, &mut output));
+        }
+    }
+    #[test]
+    fn test_starts_with_eval() {
+        let input = get_default_input();
+        let mut output = Output {
+            show: true,
+            boost: 1.0,
+            price: Default::default(),
+        };
+
+        let cases = [
+            (
+                Value::String(String::from("1234567890")),
+                Value::String(String::from("123")),
+                true,
+            ),
+            (
+                Value::String(String::from("1234567890")),
+                Value::String(String::from("456")),
+                false,
+            ),
+            (
+                Value::String(String::from("1234567890")),
+                Value::String(String::from("1234567890")),
+                true,
+            ),
+            (
+                Value::String(String::from("1234567890")),
+                Value::String(String::from("12345678901")),
+                false,
+            ),
+        ];
+
+        for (string, starting, expected) in cases.iter() {
+            let rule = Rule::Function(Function::new_starts_with(
+                Rule::Value(string.clone()),
+                Rule::Value(starting.clone()),
+            ));
+            let expected = Some(Value::Bool(*expected));
+            assert_eq!(Ok(expected), rule.eval(&input, &mut output));
+        }
+    }
+    #[test]
+    fn test_ends_with_eval() {
+        let input = get_default_input();
+        let mut output = Output {
+            show: true,
+            boost: 1.0,
+            price: Default::default(),
+        };
+
+        let cases = [
+            (
+                Value::String(String::from("1234567890")),
+                Value::String(String::from("890")),
+                true,
+            ),
+            (
+                Value::String(String::from("1234567890")),
+                Value::String(String::from("123")),
+                false,
+            ),
+            (
+                Value::String(String::from("1234567890")),
+                Value::String(String::from("1234567890")),
+                true,
+            ),
+        ];
+
+        for (string, starting, expected) in cases.iter() {
+            let rule = Rule::Function(Function::new_ends_with(
+                Rule::Value(string.clone()),
+                Rule::Value(starting.clone()),
+            ));
+            let expected = Some(Value::Bool(*expected));
+            assert_eq!(Ok(expected), rule.eval(&input, &mut output));
+        }
+    }
+
+    #[test]
+    fn test_get_dai_price_in_usd_eval() {
```

Comment:
Could you extract the strings in `const`s (`DAI`, `USDC`, `USDT`) and we can keep the list in a `HashMap` as well and thus remove the need for the whole `fn get_deposit_asset_divisor()` and use the provided methods from the `HashMap` (you would need `lazy_static` for the `HashMap`).

This way the tests get redundant